src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -276,119 +275,42 @@ class LspConnectionManager {
-  private _o = 0
-  _insertBuffer = req => {
-    this._bufferComposition[this._i % 100] = req
-    this._i++
-  }
-  _incrementBuffer = () => {
-    const msg = this._bufferComposition[this._o % 100]
-    if (!!msg) {
-      this.inject(...msg)
-      this._o++
-    }
-    return
-  }
-
-  _addUpdatesToBuffer(
+  _updateLsp(
     toAdd: Partial<CompositionSelection>,
-    toRemove: Partial<CompositionSelection>
+    toRemove: Partial<CompositionSelection> = null
   ) {
-    /* order is important. This ordering must occur on several levels:
-        (1) bucket & measurement changes must be applied first.
-        (2) for array items (fields and tagValues):
-            * remove all, before adding all from current.
-            * such that on re-sync with the session store...it does a full replacement.
-        (3) even if the Lsp received the executeCommands in order, it may not run these in order.
-            * spec is purely atomic operations, without order mattering.
-            * but since we require that the AddField etc has an init composition -- order does matter.
-            * If on page reload:
-                * we don't AddBucket before AddField --> it will fail.
-                * we AddField twice too quickly, each will see the original text as having 0 fields
-                  * therefore, each addField returns an applyEdit for 1 field
-              * solution:
-                  * short term:
-                    * buffer of executeCommands, send to Lsp at a throttled pace (hack timeouts)
-                  * longterm: middleware? changes in Lsp?
-    */
-    const numFieldChanges =
-      (toAdd.fields?.length || 0) + (toRemove.fields?.length || 0)
-    const numTagValueChanges =
-      (toAdd.tagValues?.length || 0) + (toRemove.tagValues?.length || 0)
-    const reInitBlock =
-      toAdd.bucket ||
-      toAdd.measurement ||
-      numFieldChanges + numTagValueChanges > 1
-
-    if (reInitBlock) {
-      const payload: Partial<CompositionInitParams> = {
-        bucket: toAdd.bucket?.name || this._session.bucket?.name,
-      }
-      if (toAdd.measurement || this._session.measurement) {
-        payload['measurement'] = toAdd.measurement || this._session.measurement
-      }
-      if (toAdd.fields || this._session.fields) {
-        payload['fields'] = toAdd.fields || this._session.fields
-      }
-      if (toAdd.tagValues || this._session.tagValues) {
-        payload['tagValues'] = (toAdd.tagValues || this._session.tagValues).map(
-          ({key, value}) => [key, value]
-        )
-      }
-      this._insertBuffer([ExecuteCommand.CompositionInit, payload])
-      return // re-initialize full block. no more requests needed.
+    if (toAdd.bucket) {
+      this.inject(ExecuteCommand.CompositionInit, {bucket: toAdd.bucket?.name})
+    }
+
+    if (toAdd.measurement) {
+      this.inject(ExecuteCommand.CompositionSetMeasurement, {
+        value: toAdd.measurement,
+      })
     }
 
     if (toRemove.fields?.length) {
```

Comment:
```suggestion
    if (toRemove?.fields?.length) {
```
Should there by a `?` after `toRemove`? Since it is set to `null` by default on line 280. Same for the the rest of the `toRemove`

Reply:
TY! done.